pkg/reconciler/knativeeventing/common/dispatcheradaptertransform.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	mf "github.com/manifestival/manifestival"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type unstructuredGetter interface {
+	Get(obj *unstructured.Unstructured) (*unstructured.Unstructured, error)
+}
+
+// DispatcherAdapterTransform keeps the number of replicas and the env vars, if the deployment
+// pingsource-mt-adapter, kafka-ch-dispatcher or imc-dispatcher exists in the cluster.
+func DispatcherAdapterTransform(client unstructuredGetter) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() == "Deployment" && (u.GetName() == "pingsource-mt-adapter" ||
+			u.GetName() == "imc-dispatcher") {
+			current, err := client.Get(u)
+			if errors.IsNotFound(err) {
+				return nil
+			}
+			if err != nil {
+				return err
+			}
+			return setReplicaEnvVars(u, current)
+		}
+		return nil
+	}
+}
+
+func setReplicaEnvVars(u, current *unstructured.Unstructured) error {
+	numReplicas, found := nestedInt64OrFloat64(current.Object, "spec", "replicas")
+	if found {
+		if err := unstructured.SetNestedField(u.Object, numReplicas, "spec", "replicas"); err != nil {
+			return err
+		}
+	}
```

Comment:
Why are we working with unstructured values throughout? That feels like it makes things unnecessarily hard to follow (and contains quite a few unchecked casts etc.).

I experimented with converting into actual deployments a bit, see https://github.com/knative/operator/commit/bfbcc9cb34bcec67682b5f4409b00c6bec4bce45

Reply:
Thx.  I will change it.